blog/2023-07-23-openhab-4-0-release.md

```diff
@@ -0,0 +1,354 @@
+A replacement for the Nashorn JS is **GraalJS** ("the holy grail"), which is currently running **ECMAScript 2022** and therefore supports all modern JavaScript features, like arrow functions and private class fields and methods.
+[**GraalJS**](https://github.com/oracle/graaljs) is already available in openHAB 3 when the [JavaScript Scripting add-on](/addons/automation/jsscripting/) is installed.
+The [JavaScript Scripting add-on](/addons/automation/jsscripting/) also includes the [openHAB JavaScript library (openhab-js)](https://github.com/openhab/openhab-js), which in fact empowers most of the new Blockly features and allows the much cleaner code by providing a simple, pure JS API to openHAB.
+
+**New Features in Blockly**
+
+* Full Support of **U**nits **o**f **M**easurement with [5 brand-new blocks](/docs/configuration/blockly/rules-blockly-uom.html)
+* The [Math section](/docs/configuration/blockly/rules-blockly-standard-ext.html#math) was extended by a new bitshift block and extended rounding block
+* Metadata blocks allow direct access to [Item metadata](/docs/configuration/blockly/rules-blockly-items-things.html#item-metadata)
+* A historicState option was added to the [persistence blocks](/docs/configuration/blockly/rules-blockly-persistence.html#get-statistical-value-of-an-item)
+* Persistence blocks can now be used with any [persistence service](/docs/configuration/blockly/rules-blockly-persistence.html)
+* Added pattern to [text of date block](/docs/configuration/blockly/rules-blockly-date-handling.html#get-string-representation-of-date-text-of)
+* Support for [private and global cache](/docs/configuration/blockly/rules-blockly-value-storage.html#caching) when using the value storage blocks
+* A new [Blockly dictionary loop](/docs/configuration/blockly/rules-blockly-standard-ext.html#loop-over-a-dictionary) block with value and key from dictionary
+* Added stricter type checking for blocks to avoid mistakes
+* Added numeric state, quantity state to getItemAttribute
+
+Editor:
+* [Search](/docs/configuration/blockly/rules-blockly-before-using.html#searching-the-workspace) within the blockly workspace via Cmd/Ctrl-F
+* Shadow Blocks are now automatically converted to real blocks, which is good when you want to reuse them by duplicating the block 
+* [Cross-Rule Copy & Paste](https://next.openhab.org/docs/configuration/blockly/rules-blockly-before-using.html#cross-rule-copy-paste)
+
+All new blocks have received comprehensive documentation at the [Blockly Reference](/docs/configuration/blockly/).
+
+### Scene Editor
+
+Given several discussions on the community, support for scenes was a long awaited feature — openHAB 4 has it.
+
+Basically, scenes are only a set of Items and states that the Items are commanded to when the scene is enabled.
+In openHAB, a simple rule would be enough to achieve that functionality, but it would either require some coding or several UI interactions.
```

Comment:
```suggestion
In openHAB, a simple rule would be enough to achieve this functionality, but it would either require some coding or several UI interactions.
```